### PR TITLE
Issue 75 bug frontend fields should match backend

### PR DESCRIPTION
**Pull request by crawsoir**
- Replace old event fields to match backend
- Replace all `building` fields to be JS objects
- Rewrite `EventData.js` and `LocationData.js` to be acceptable for the backend
- Modify the calendar page to handle the calendar component requiring the old fields

### CODE REVIEW

File: utmap-client/src/components/pages/CalendarPage.js [hunk 1]
```diff
@@ -80,6 +106,7 @@ const groupEvents = (eventsList) => {
 	//Groups events on the same day together if there are more than 3
 	let count = 0;
 	return eventsList.reduce((newList, event) => {
+		event = convertToCalendarEvent(event);
 		let lastElem = newList[newList.length - 1];
 		if(newList.length === 0 || 
 			!isSameDay(Date.parse(event.startDate), Date.parse(lastElem.startDate))) {
```
Comment (RiddleMeDoo) [inline utmap-client/src/components/pages/CalendarPage.js:112]: Need to change it here

Comment (crawsoir) [reply utmap-client/src/components/pages/CalendarPage.js:112]: Just to clarify, do you mean to convert `event` back to a UTMap event? Unless I'm misunderstanding!

Comment (RiddleMeDoo) [reply utmap-client/src/components/pages/CalendarPage.js:112]: Oops, I commented on the wrong line. See line 84/112, event.startDate

Comment (RiddleMeDoo) [inline utmap-client/src/components/pages/CalendarPage.js:112]: This line has event.startDate

**Review by RiddleMeDoo — APPROVED** [review]

**Review by ShahmeerShahid — DISMISSED** [review]

**Comment by crawsoir** [comment]
deprecated

**Pull request closed without merging**